src/frontend/src/utils/iiConnection.ts

```diff
@@ -272,22 +272,16 @@ export class Connection {
     };
   };
 
-  private updateKeyTypeIfNecessary(
+  private updateOriginIfNecessary(
     devices: Omit<DeviceData, "alias">[],
-    attachmentInfo: {
-      credentialId: ArrayBuffer;
-      authenticatorAttachment: AuthenticatorAttachment;
-    },
+    pubkey: ArrayBuffer,
     authenticatedConnection: AuthenticatedConnection
   ) {
-    // Find the device based on the credential_id
-    const device = findDeviceByCredentialId(
-      devices,
-      attachmentInfo.credentialId
-    );
+    // Find the device based on the pubkey
+    const device = findDeviceByPubkey(devices, pubkey);
 
-    // only update devices with key-type unknown
-    if (device !== undefined && "unknown" in device.key_type) {
+    // only update devices without origin information
+    if (device !== undefined && device.origin.length === 0) {
```

Comment:
```suggestion
    if (device !== undefined && device.origin.length === 0) {
              device.origin satisfies string[];
```

Would add that just in case we ever fix the generated interface, in which case `origin` would be a `string` but `length` would still work